src/org/parosproxy/paros/control/Control.java

```diff
@@ -546,8 +546,10 @@ public Mode getMode() {
 		return mode;
 	}
 	public void setMode(Mode mode) {
-		this.mode = mode;
-		getExtensionLoader().sessionModeChangedAllPlugin(mode);
-		model.getOptionsParam().getViewParam().setMode(mode.name());
+		if (!mode.name().equals(model.getOptionsParam().getViewParam().getMode())) {

```

Comment:
Something strange seems to be going on with github. 
> Add more commits by pushing to the rajiv65-develop branch on kingthorin/zaproxy.

Which is what I did, and if you visit it (https://github.com/kingthorin/zaproxy/tree/rajiv65-develop) you can see 
> Latest commit 41967f0 

for some reason it doesn't seem to be being reflected in the PR.